src/vswhere.lib/CommandArgs.cpp

```diff
@@ -231,6 +234,37 @@ void CommandArgs::Usage(_In_ Console& console) const
     }
 }
 
+std::wregex CommandArgs::ParseRegex(_In_ const std::wstring& pattern) noexcept
+{
+    // Reserve ~125% of the incoming pattern to hold any changes.
+    wstring accumulator;
+    accumulator.reserve(pattern.size() * 1.25);
+
+    for (auto it = pattern.begin(); it != pattern.end(); ++it)
+    {
+        switch (*it)
+        {
+        case L'.':
+            accumulator += L"\\.";
+            break;
+
+        case L'*':
+            accumulator += L".*";
+            break;
+
+        case L'?':
+            accumulator += L".";
+            break;
+
+        default:
```

Comment:
Anything else we should escape? I think this is all we should have in package IDs that is a regex special char so we are probably good

Reply:
A literal `.` is all I could think about to escape, and `*` and `?` are all the wildcards we'd support that DOS does, so I think we're fine here. I couldn't think of anything else in package IDs that would be allowed e.g., `(`, `{`, `+`, etc., that would need to be escaped. IIRC, we only allowed alphanumeric + `.`, right?

Reply:
Looking at all IDs in a VS2022 catalog, I see nothing else we need to escape. For a breakdown:

```powershell
$catalog = gc <path> | convertfrom-json
$catalog.packages.id | foreach { $_.GetEnumerator() } | group -noelement | sort count -desc
```
```text
Count Name
----- ----
72586 .
47409 o
44286 s
35784 e
30246 t
27549 r
26777 i
18999 c
18386 a
17515 u
16168 M
12666 1
11854 l
11776 4
11529 f
11359 R
11282 C
11018 V
 9409 T
 8418 6
 8102 d
 7036 g
 6684 S
 6478 n
 5313 b
 5227 3
 5111 m
 4696 A
 4337 7
 3850 p
 3719 X
 3559 H
 2887 P
 2782 8
 2560 x
 2321 D
 2251 0
 2220 2
 1429 5
 1388 k
 1139 I
 1076 B
 1067 y
  992 W
  952 E
  876 F
  811 v
  795 N
  793 L
  756 h
  682 9
  662 O
  469 w
  376 G
  316 U
  198 j
  191 K
  183 z
  162 -
   60 q
   38 Q
   34 _
   28 J
```